HACKING.md

```diff
@@ -54,6 +54,44 @@ quite pleasant.
 To deal with process multiplexing, Dune uses a simplified
 Lwt/Async-like monad, implemented in [src/fiber/fiber.mli](src/fiber/fiber.mli).
 
+## Tests
+
+Dune uses [cram style](https://blog.janestreet.com/testing-with-expectations/)
+tests for its test suite. The test suite is contained in
+[test/blackbox-tests](test/blackbox-tests). A single test consists of a test
+directory in the test-cases/ sub directory which contains a run.t file defining
+the test.
+
+An example `run.t` file:
+
+```
+A description of the test. The command running the tests is preceeded by
+two spaces and a $. The expected output of the command is also indetended by
+two spaces and is right below the command (note the absence of a $)
+
+  $ echo "the expected output is below"
+  the expected output is below
+```
+
+Running the entire test suite is done with `$ make test`. A particular test can
+be executed with `$ ./boot.exe build @<test-name>`.
```

Comment:
boot.exe doesn't support that. I wonder if we shouldn't just recommend to call `dune build ...` rather that give custom/make commands. It seems better to encourage to use dune directly.

Reply:
Ah yeah, indeed. I actually just use an alias for the dev version of dune myself. Does it make sense to just add a `$ make dune <args>` target that will delegate to dune?

Reply:
I was thinking of just using an installed dune binary

Reply:
That also has some disadvantages. Like having to make sure the dune binary is installed. But ok, I'll change it to that. I'll keep the new make target if you don't mind.